freedomofpress/kernel-builder
# Updates support for e1000e network hardware

### Status
WIP, patches are not applying cleanly yet. More work required.

### Description  
We've long been supporting the e1000e chipset for Intel NUCs. In pursuit of support for the NUC10, we'll need to pull in more recent patches to the e1000e from the upstream kernel. To wit:

  0. git clone https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
  1. git checkout v5.10
  2. git log ./drivers/net/ethernet/intel/e1000e/hw.h
  3. inspect output, extract commit hashes
  4. git format-patch -1 <sha> for each
  5. rename .patch files with numeric prefix, for reliable sorting
    
These changes aren't tested yet, but based on the content of the patches, these should bring our custom kernel up to speed with NUC10 support.
    
Tweaked the patch logic to ensure that patches are applied in numeric-sort order, and renamed the oldest patches to accommodate.

### Testing
To make the same packages yourself:

1. Make sure you have credentials exported as `GRSECURITY_{USER,PASSWORD}`
2. Run `make securedrop-core`
3. Wait.
4. View artifacts in `build/`
